In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [ ]:
import kagglehub

path = kagglehub.dataset_download("irakozekelly/pharmaceutical-r-and-d-clinical-trial-data-overview")

print("Path to dataset files:", path)

Using Colab cache for faster access to the 'pharmaceutical-r-and-d-clinical-trial-data-overview' dataset.
Path to dataset files: /kaggle/input/pharmaceutical-r-and-d-clinical-trial-data-overview


In [ ]:
import pandas as pd

df = pd.read_csv("/content/AERO-BirdsEye-Data.csv")

print(df.head())
print(df.info())
print(df.describe())

                      NCT                                              Title  \
0  Sanofi/NCT00003305.xml  A Phase II Trial of Aminopterin in Adults and ...   
1  Sanofi/NCT00003821.xml  Phase II Trial of Aminopterin in Patients With...   
2  Sanofi/NCT00004025.xml  Phase I/II Trial of the Safety, Immunogenicity...   
3  Sanofi/NCT00005645.xml  Phase II Trial of ILX295501 Administered Orall...   
4  Sanofi/NCT00008281.xml  A Multicenter, Open-Label, Randomized, Three-A...   

                                             Summary  Start_Year  Start_Month  \
0  RATIONALE: Drugs used in chemotherapy use diff...        1997            7   
1  RATIONALE: Drugs used in chemotherapy use diff...        1998            1   
2  RATIONALE: Vaccines made from a person's white...        1999            3   
3  RATIONALE: Drugs used in chemotherapy use diff...        1999            5   
4  RATIONALE: Drugs used in chemotherapy use diff...        2000           10   

             Phase              

In [ ]:
#problem 1
df = df.dropna(subset=["Phase"])
df["Phase"] = df["Phase"].str.strip()

df.groupby("Phase")["Enrollment"].mean().sort_values()


,Enrollment
Phase,
Phase 1,51.106916
Early Phase 1,58.200000
Phase 1/Phase 2,99.860248
Phase 2,180.114850
Phase 4,518.169231
Phase 2/Phase 3,564.278571
Phase 3,795.794762


The later the phases the higher enrollment across clinical trials.
Later phases of trials show significantly higher enrollment, which could indicate either increased operation scale or resource requirements in advance stages of development. Phase 3 shows the highest enrollment rate out of all.


"Enrollment size increases substantially across clinical trial phases, with Phase 3 trials averaging nearly 800 participants compared to approximately 50 in Phase 1. This reflects the growing scale and resource requirements of later-stage trials, where broader populations are needed to validate efficacy and safety."

In [ ]:
print(df["Enrollment"].describe())

count    13486.000000
mean       431.645707
std       1802.505789
min          0.000000
25%         40.000000
50%        124.000000
75%        368.750000
max      84496.000000
Name: Enrollment, dtype: float64


distribution: mean is higher than min, which means skewed data

“Enrollment data is highly skewed, with a small number of very large trials significantly increasing the average. This suggests that while most trials operate at moderate scale, a few large-scale studies drive overall resource demands.”

In [ ]:
#problem 3
top_conditions = df["Condition_Final"].value_counts().head(10)
print(top_conditions)

Condition_Final
Diabetes Mellitus, Type 2                 529
Breast Neoplasms                          388
Hypertension                              338
Pulmonary Disease, Chronic Obstructive    336
Asthma                                    331
Arthritis, Rheumatoid                     329
Influenza, Human                          323
Schizophrenia                             290
Diabetes Mellitus                         262
Alzheimer Disease                         214
Name: count, dtype: int64


The highest volume activity regarding the conditions tested within the clinical trial is the condition Diabetes Mellitus, Type 2.

“Clinical trial activity is heavily concentrated in chronic and high-prevalence conditions such as Type 2 diabetes, breast cancer, and hypertension. This suggests that pharmaceutical R&D efforts are strategically aligned with diseases that represent significant global health burdens and commercial opportunities.”

In [ ]:
#problem 2
status_counts = df["Status"].value_counts(normalize=True) * 100
print(status_counts)

Status
Completed                  76.753670
Terminated                  9.380098
Recruiting                  5.843097
Active, not recruiting      4.753077
Withdrawn                   2.120718
Not yet recruiting          0.793415
Unknown status              0.126057
Suspended                   0.118642
Enrolling by invitation     0.111226
Name: proportion, dtype: float64


In [ ]:
pd.crosstab(df['Phase'], df['Status'], normalize='index') * 100
#print("Completed")
#print("Terminated")

Status,"Active, not recruiting",Completed,Enrolling by invitation,Not yet recruiting,Recruiting,Suspended,Terminated,Unknown status,Withdrawn
Phase,,,,,,,,,
Early Phase 1,0.000000,100.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
Phase 1,3.696343,78.577107,0.039746,0.874404,6.637520,0.317965,8.068362,0.039746,1.748808
Phase 1/Phase 2,12.111801,54.037267,0.310559,0.931677,18.012422,0.000000,11.801242,0.621118,2.173913
Phase 2,4.310345,73.581758,0.083426,1.001112,5.450501,0.139043,12.486096,0.027809,2.919911
Phase 2/Phase 3,5.000000,64.285714,0.714286,0.000000,12.857143,0.000000,16.428571,0.000000,0.714286
Phase 3,6.507060,77.655003,0.184162,0.716186,5.627174,0.040925,7.550645,0.184162,1.534684
Phase 4,1.439206,82.332506,0.000000,0.545906,3.672457,0.049628,9.081886,0.198511,2.679901


76.75% of clinical trials are completed while 9.38% are terminated showing that the majority of clinical trials across the board are adequately following requirements and regulations. (mangerial insight)

“Approximately 77% of clinical trials reach completion, while nearly 10% are terminated early. This highlights a meaningful level of operational and financial risk in the R&D pipeline, emphasizing the importance of effective trial design and resource allocation.”

In [ ]:
status_by_phase = pd.crosstab(df['Phase'], df['Status'], normalize='index') * 100
print(status_by_phase[['Completed', 'Terminated']])

Status            Completed  Terminated
Phase                                  
Early Phase 1    100.000000    0.000000
Phase 1           78.577107    8.068362
Phase 1/Phase 2   54.037267   11.801242
Phase 2           73.581758   12.486096
Phase 2/Phase 3   64.285714   16.428571
Phase 3           77.655003    7.550645
Phase 4           82.332506    9.081886


“Completion rates vary across clinical trial phases, with Phase 4 and Phase 3 trials demonstrating the highest likelihood of completion. In contrast, transitional phases such as Phase 1/Phase 2 and Phase 2/Phase 3 exhibit notably lower completion rates and higher termination rates. This suggests that these intermediate stages represent critical decision points in the clinical development process, where trials face increased scientific and operational uncertainty. From a managerial perspective, this highlights the importance of careful evaluation and resource allocation during transitional phases, as they carry a higher risk of failure. These findings indicate that while later-stage trials require greater investment due to larger enrollment sizes, the greatest risk of failure occurs earlier in the development pipeline.”